.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,25 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
These should always be pinned to a hash to avoid compromised deps.

https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash

Reply:
Good catch.  I filed https://github.com/golang/geo/issues/142 which, AFAIU, includes a check for this.

Reply:
Came from https://github.com/marketplace/actions/golangci-lint#how-to-use the Simple Example

Reply:
Do you know where to find the hash to use? 

https://github.com/actions/checkout/releases  If i click on 4.2.2 the current latest release, I see 
https://github.com/actions/checkout/commit/11bd71901bbe5b1630ceea73d27597364c9af683  which I am guessing is the hash I want to use.

Reply:
https://github.com/actions/checkout/commit/11bd71901bbe5b1630ceea73d27597364c9af683 seems right to me.  It's the last commit here:
https://github.com/actions/checkout/compare/v4.2.1...v4.2.2

Reply:
Done